File: Java/src/main/java/hayaa/rpc/common/RpcDataHelper.java
package hayaa.rpc.common;

import hayaa.common.JsonHelper;
import hayaa.rpc.common.protocol.RpcDataValue;
import hayaa.rpc.server.ServerClassDataStore;

public class RpcDataHelper {

    public static int parseDataType(Object arg) {
        Class<?> clazz = arg.getClass();
        int result = 1;
        switch (clazz.getName()) {
            case "java.lang.String":
                result = 2;
                break;
            case "java.lang.Integer":
                result = 3;
                break;
            case "java.lang.Float":
                result = 4;
                break;
            case "java.lang.Double":
                result = 5;
                break;
            default:
                break;
        }
        return result;
    }

    /**
     * 服务端参数数据转换
     *
     * @param rpcDataValue
     * @return
     */
    public static Object parseServerData(RpcDataValue rpcDataValue) {
        Object result = null;
        Class classObj = ServerClassDataStore.getClass(rpcDataValue.getClassName());
        if(classObj!=null){
            result=JsonHelper.gsonDeserialize(rpcDataValue.getValContainer(),classObj);
        }
        return result;
    }
}

File: C#/RPCV2/Hayaa.RPC.Common/Util/RpcDataHelper.cs
using Hayaa.RPC.Common.Protocol;
using Hayaa.RPC.Service.Protocol;
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.RPC.Service.Util
{
  public  class RpcDataHelper
    {
        /// <summary>
        /// 获取参数变量的多平台通用数据类型
        /// </summary>
        /// <param name="arg">方法参数变量的类型</param>
        /// <returns></returns>
        public static int ParseDataType(Type type)
        {
            int result = CommunicationPrimitives.CLASS;
            switch (type.FullName)
            {
                case "System.String":
                    result = CommunicationPrimitives.String;
                    break;
                case "System.Boolean":
                    result = CommunicationPrimitives.Boolean;
                    break;
                case "System.Byte":
                    result = CommunicationPrimitives.Byte;
                    break;
                case "System.Int16":
                    result = CommunicationPrimitives.Short;
                    break;
                case "System.Int32":
                    result = CommunicationPrimitives.Integer;
                    break;
                case "System.Int64":
                    result = CommunicationPrimitives.Long;
                    break;
                case "System.Single":
                    result = CommunicationPrimitives.Float;
                    break;
                case "System.Double":
                    result = CommunicationPrimitives.Double;
                    break;
                case "System.Decimal":
                    result = CommunicationPrimitives.Decimal;
                    break;
                default:
                    break;
            }
            return result;
        }
       
        public static T ParseValueTypeData<T>(String valContainer,out Boolean isClass)
        {
            isClass = false;
            Object result = null;
            String name = typeof(T).FullName;
            switch (name)
            {
                case "System.String":
                    result = valContainer;
                    break;
                case "System.Boolean":
                    result = Boolean.Parse(valContainer);
                    break;
                case "System.Byte":
                    result = Byte.Parse(valContainer);
                    break;
                case "System.Int16":
                    result = short.Parse(valContainer);
                    break;
                case "System.Int32":
                    result = int.Parse(valContainer);
                    break;
                case "System.Int64":
                    result = long.Parse(valContainer);
                    break;
                case "System.Single":
                    result = float.Parse(valContainer);
                    break;
                case "System.Double":
                    result = double.Parse(valContainer);
                    break;
                case "System.Decimal":
                    result = decimal.Parse(valContainer);
                    break;
                default:
                    break;
            }
            if (result == null) {
                isClass = true;
                return default(T); }
            return (T)result;
        }
    }
}
